src/test/java/mate/academy/bookstore/CategoryServiceTest.java

```diff
@@ -0,0 +1,195 @@
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.testcontainers.shaded.org.apache.commons.lang3.builder.EqualsBuilder;
+
+@ExtendWith(MockitoExtension.class)
+public class CategoryServiceTest {
+    @Mock
+    private CategoryRepository categoryRepository;
+    @Mock
+    private CategoryMapper categoryMapper;
+    @InjectMocks
+    private CategoryServiceImpl categoryService;
+
+    @DisplayName("Find all categories")
+    @Test
+    void findAll_ValidCategoryDto_Success() {
+        // given
+        Category category = getCategoryA();
+        CategoryResponseDto categoryResponseDto = getCreateCategoryResponseDto();
+        Pageable pageable = PageRequest.of(0, 10);
+        List<Category> categories = List.of(category);
+        Page<Category> categoryPage = new PageImpl<>(
+                categories, pageable, categories.size());
+        Mockito.when(categoryRepository.findAll(pageable)).thenReturn(categoryPage);
+        Mockito.when(categoryMapper.toDto(category)).thenReturn(categoryResponseDto);
+        // when
+        List<CategoryResponseDto> actual = categoryService.findAll(pageable);
+        // then
+        Assertions.assertEquals(1, actual.size());
+        CategoryResponseDto dto = actual.get(0);
+        Assertions.assertNotNull(dto);
+        EqualsBuilder.reflectionEquals(categoryResponseDto, dto);
+    }
+
+    @Test
+    @DisplayName("Find all empty category list")
+    void findAll_EmptyCategoryList_False() {
+        // given
+        Pageable pageable = PageRequest.of(0, 10);
+        Page<Category> emptyCategoryPage = new PageImpl<>(Collections.emptyList(), pageable, 0);
+        when(categoryRepository.findAll(pageable)).thenReturn(emptyCategoryPage);
+        // when
+        List<CategoryResponseDto> actual = categoryService.findAll(pageable);
+        // then
+        assertTrue(actual.isEmpty());
+    }
+
+    @Test
+    @DisplayName("Find an existing category by valid ID")
+    void getById_WithValidCategoryId_ShouldReturnValidCategory() {
+        // given
+        Category category = getCategoryA();
+        Long categoryId = category.getId();
+        CategoryResponseDto expected = getCreateCategoryResponseDto();
+
+        Mockito.when(categoryRepository.findById(anyLong()))
+                .thenReturn(Optional.of(category));
+        Mockito.when(categoryMapper.toDto(category))
+                .thenReturn(expected);
+        // when
+        CategoryResponseDto actual = categoryService.getById(categoryId);
+        // Then
+        assertEquals(expected, actual);
+    }
+
+    @DisplayName("Get category by non existing ID")
+    @Test
+    void getById_WithNonExistingId_ShouldThrowException() {
+        // given
+        Long testId = 100L;
+        Mockito.when(categoryRepository.findById(testId))
+                .thenThrow(new EntityNotFoundException("Can't find category by id: "
+                        + testId));
```

Comment:
Looks weird, repository can return empty optional, we would throw this exception on service layer